client/transaction-pool/src/lib.rs

```diff
@@ -727,6 +774,220 @@ where
 	}
 }
 
+/// Trait for deciding if core part of maintenance procedure shall be executed.
+///
+/// For the following chain:
+///   B1-C1-D1-E1
+///  /
+/// A
+///  \
+///   B2-C2-D2-E2
+///
+/// Some scenarios and expected behavior:
+/// nbb(C1), f(C1) -> false (handle_enactmanet was already performed in nbb(C1))
+/// f(C1), nbb(C1) -> false (handle_enactmanet was already performed in f(C1))
+///
+/// f(C1), nbb(D2) -> false (handle_enactmanet was already performed in f(C1), we should not retract
+/// finalized block) f(C1), f(C2), nbb(C1) -> false
+///
+/// nbb(C1), nbb(C2) -> true (switching fork is OK)
+pub trait EnactmentPolicy<Block: BlockT> {
```

Comment:
I did it to access `enactment_helper` API in tests. I wanted to test the API directly, and didn't want to make `EnactmentHelper` struct public.

Reply:
Why do we need this in tests at all? This entire struct could probably also be just some function if you switch to use `tree_route` function directly.

Reply:
The logic of this function is more complex than I initially thought. It can be easily broken, and I decided to cover it with some unit tests.

I am not opting for trait - maybe there is more elegant way to expose this component to tests module.